modules/maven/src/test/java/org/eclipse/sw360/antenna/maven/workflow/processors/MavenArtifactResolverTest.java

```diff
@@ -0,0 +1,136 @@
+        when(antennaContextMock.getGeneric(MavenProject.class))
+                .thenReturn(Optional.of(mavenProject));
+        LegacySupport legacySupport = null;
+        when(antennaContextMock.getGeneric(LegacySupport.class))
+                .thenReturn(Optional.ofNullable(legacySupport));
+
+        when(configMock.getIgnoreForSourceResolving()).thenReturn(Collections.emptyList());
+        when(project.getBasedir()).thenReturn(new File(""));
+        when(toolConfigMock.getAntennaTargetDirectory()).thenReturn(temporaryFolder.newFolder("target").toPath());
+        Path targetDir = toolConfigMock.getAntennaTargetDirectory();
+        when(toolConfigMock.getDependenciesDirectory()).thenReturn(targetDir.resolve("dependencies"));
+
+        mavenArtifactResolver = new MavenArtifactResolver();
+        mavenArtifactResolver.setAntennaContext(antennaContextMock);
+
+        artifacts = Collections.singletonList(mkArtifact("commons-csv", "org.apache.commons", "1.7"));
+    }
+
+    @Test
+    public void mavenArtifactResolverForGoodArtifact() throws AntennaException {
+        Map<String, String> configMap = new HashMap<>();
+        configMap.put("sourceRepositoryUrl", "");
+        configMap.put("preferredSourceQualifier", "sources-ext");
+        mavenArtifactResolver.configure(configMap);
+
+        Collection<Artifact> processedArtifacts = mavenArtifactResolver.process(artifacts);
+
+        assertThat(processedArtifacts.size()).isEqualTo(1);
+        assertThat(antennaContextMock.getToolConfiguration().getDependenciesDirectory().resolve("commons-csv-1.7.jar")).exists();
+    }
+
+    @Test
+    public void mavenArtifactResolverForGoodArtifactDirectWithImpl() throws MalformedURLException, AntennaException {
+        ProxySettings proxySettings = new ProxySettings(false, null, 0);
+
+        ContextExtension contextExtension = new ContextExtension();
+
+        Collection<Artifact> processedArtifacts = new MavenArtifactResolverImpl(proxySettings, contextExtension.get(RepositorySystem.class),
+                antennaContextMock.getGeneric(MavenProject.class), contextExtension.get(LegacySupport.class),
+                toolConfigMock.getDependenciesDirectory(), Collections.emptyList(), null,
+                null, reporterMock, false, antennaContextMock.getProject().getBasedir())
+                .process(artifacts);
+
+        assertThat(processedArtifacts.size()).isEqualTo(1);
+        assertThat(antennaContextMock.getToolConfiguration().getDependenciesDirectory().resolve("commons-csv-1.7.jar")).exists();
+    }
+
+    private Artifact mkArtifact(String groupId, String name, String version) {
+        Artifact artifact = new Artifact();
+
+        artifact.addFact(new MavenCoordinates(groupId, name, version));
+        artifact.addFact(new ArtifactMatchingMetadata(MatchState.EXACT));
+
+        return artifact;
+    }
+
+    @Override
+    @After
+    public void assertThatOnlyExpectedMethodsAreCalled() {
```

Comment:
this is usually handled differently, we do not overwrite `assertThatOnlyExpectedMethodsAreCalled` anywhere else (I think).

Reply:
https://github.com/eclipse/antenna/blob/fc457dd0829e0dfe5aad2bd151bafe1abadba8e1/core/core-workflow-steps/src/test/java/org/eclipse/sw360/antenna/workflow/analyzers/CsvAnalyzerTest.java#L148-150

but I'll correct it anyways